src/setup/node.rs

```diff
@@ -0,0 +1,167 @@
+
+        self.meta.start_args.push("-d".into());
+        self.meta.start_args.push(target.into());
+
+        if self.conf.log_to_stdout {
+            self.meta.start_args.push("-o".into());
+        }
+
+        let node = self.start_node();
+
+        // TODO(Rqnsom) wait for the connection to confirm
+
+        self.meta = NodeMetaData::new(setup_path)?; // Reset args
+        Ok(node)
+    }
+
+    /// Sets whether to log the node's output to Ziggurat's output stream.
+    pub fn log_to_stdout(mut self, log_to_stdout: bool) -> Self {
+        self.conf.log_to_stdout = log_to_stdout;
+        self
+    }
+
+    fn start_node(&self) -> Node {
+        let (stdout, stderr) = match self.conf.log_to_stdout {
+            true => (Stdio::inherit(), Stdio::inherit()),
+            false => (Stdio::null(), Stdio::null()),
+        };
+
+        let child = Command::new(&self.meta.start_command)
+            .current_dir(&self.meta.path)
+            .args(&self.meta.start_args)
+            .stdin(Stdio::null())
+            .stdout(stdout)
+            .stderr(stderr)
+            .spawn()
+            .expect("Node failed to start");
+
+        Node {
+            child,
+            config: self.conf.clone(),
+        }
+    }
+}
+
+pub struct Node {
+    child: Child,
+    config: NodeConfig,
+}
+
+impl Node {
+    pub fn builder() -> NodeBuilder {
+        NodeBuilder::new()
+            .map_err(|e| format!("Unable to create a builder: {:?}", e))
+            .unwrap()
+    }
+
+    pub fn stop(&mut self) -> io::Result<ChildExitCode> {
+        match self.child.try_wait()? {
+            None => self.child.kill()?,
+            Some(code) => return Ok(ChildExitCode::ErrorCode(code.code())),
+        }
+        let exit = self.child.wait()?;
+
+        match exit.code() {
+            None => Ok(ChildExitCode::Success),
+            Some(exit) if exit == 0 => Ok(ChildExitCode::Success),
+            Some(exit) => Ok(ChildExitCode::ErrorCode(Some(exit))),
+        }
+    }
+}
+
+impl Drop for Node {
+    fn drop(&mut self) {
+        // We should avoid a panic.
+        if let Err(e) = self.stop() {
+            eprintln!("Failed to stop the node: {}", e);
+        }
+    }
+}
+
+fn get_node_path(node_dir_idx: usize) -> io::Result<PathBuf> {
+    Ok(get_algorand_work_path()?
+        .join(PRIVATE_NETWORK_DIR)
+        .join(format!("{NODE_DIR}{node_dir_idx}")))
+}
+
+#[cfg(test)]
+mod test {
+    use tempfile::TempDir;
+    use tokio::time::{sleep, Duration};
+
+    use super::*;
+
+    const SLEEP: Duration = Duration::from_secs(1);
+
+    #[tokio::test]
+    async fn start_stop_the_node() {
+        let builder = Node::builder();
+        let target = TempDir::new().expect("Can't build a tmp dir");
+
+        let mut node = builder
+            .log_to_stdout(true)
+            .start(target.path())
+            .await
+            .expect("Unable to start the node");
+
+        sleep(SLEEP).await;
+
+        node.stop().unwrap();
```

Comment:
```suggestion
```
no need for this one to be called explicitly, as it's triggered automatically via `Drop`